### PR TITLE
Add Classes and Interfaces for Edit FI

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Add `createEditAuthRequest()` and `edit()` stub functions to `PayPalClient`
 - Add `PayPalVaultEditAuthCallback`, `PayPalVaultEditAuthRequest`, `PayPalVaultEditAuthRequestParams`, `PayPalVaultEditAuthResult`, `PayPalVaultEditAuthResultInfo`, `PayPalVaultEditCallback`, `PayPalVaultEditRequest`, and `PayPalVaultEditResult`

### Checklist

 - [X] Added a changelog entry
 - [ ] Relevant test coverage

### CODE REVIEW

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java [hunk 1]
```diff
@@ -281,6 +286,23 @@ public void tokenize(@NonNull PayPalPaymentAuthResult.Success paymentAuthResult,
         }
     }
 
+    @ExperimentalBetaApi
+    public void createEditAuthRequest(
```
Comment (jaxdesmarais) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java:290]: Do we want to go ahead and add docstrings for the public method?

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java [hunk 2]
```diff
@@ -281,6 +286,23 @@ public void tokenize(@NonNull PayPalPaymentAuthResult.Success paymentAuthResult,
...
+        @NonNull final Context context,
+        @NonNull final PayPalVaultEditRequest payPalVaultEditRequest,
+        @NonNull final PayPalVaultEditAuthCallback payPalVaultEditAuthCallback
+    ) {
+        // TODO: implement function
+    }
+
+    @ExperimentalBetaApi
+    public void edit(
```
Comment (jaxdesmarais) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java:299]: Same here on adding docstrings

File: PayPal/src/main/java/com/braintreepayments/api/paypal/vaultedit/PayPalVaultEditRequest.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.braintreepayments.api.ExperimentalBetaApi
+
+/**
+ *
+ * Note: **This feature is in beta. It's public API may change in future releases.**
+ */
+@ExperimentalBetaApi
+data class PayPalVaultEditRequest(
+    val encryptedBillingAgreementID: String
```
Comment (scannillo) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/vaultedit/PayPalVaultEditRequest.kt:11]: We landed on `edit_paypal_vault_id` per our convos in Slack (#bt-gw-sdk-collaboration)

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java (hunk 1 above)
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.java:302]: Yep, I forgot to add them! I just pushed the changes.

File: PayPal/src/main/java/com/braintreepayments/api/paypal/vaultedit/PayPalVaultEditRequest.kt (hunk 3 above)
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/paypal/vaultedit/PayPalVaultEditRequest.kt]: thanks for calling this out! i just updated the property name.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # Braintree Android SDK Release Notes
 
 ## unreleased
+* PayPal
+    * Add PayPal edit funding instrument flow (BETA)
```
[the remark this answers is not in the document]
Comment (tdchow) [reply CHANGELOG.md]: I agree, we should stay consistent on how we refer to the feature.

We may want to update the same on iOS: https://github.com/braintree/braintree_ios/blob/f49c3a2daa4f548c06ab2e1cee46d0fd3f6d3713/CHANGELOG.md?plain=1#L8

**Review by jaxdesmarais — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jaxdesmarais) [reply CHANGELOG.md]: Sorry didn't see this before I merged the iOS PR. Updated here: https://github.com/braintree/braintree_ios/pull/1376

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**